src/os/bluestore/BlueStore.cc

```diff
@@ -9968,6 +9968,9 @@ int BlueStore::_decompress(bufferlist& source, bufferlist* result)
     _set_compression_alert(false, alg_name);
     r = -EIO;
   } else {
+    if(alg == Compressor::COMP_ALG_ZLIB) {
```

Comment:
I'd like to encourage you not to use any compression algorithm's specific logic out of relevant code base (i.e. compressor plugin / wrapper).
Suggest to introduce an additional 'opaque' parameter passed to/from of all compressors and preserved in  bluestore compression header (and its counterpart in RGW code) instead.
And extend Compressor's compress/decompress API  accordingly.